sources/lib/structures/ty.mli

```diff
@@ -26,77 +26,204 @@
 (*                                                                            *)
 (******************************************************************************)
 
+(** Types
+
+    This module defines the representation of types. *)
+
+(** {2 Definition} *)
+
 type t =
   | Tint
+  (** Integer numbers *)
   | Treal
+  (** Real numbers *)
   | Tbool
+  (** Booleans *)
   | Tunit
+  (** The unit type *)
   | Tvar of tvar
+  (** Type variables *)
   | Tbitv of int
+  (** Bitvectors of a given length *)
   | Text of t list * Hstring.t
+  (** Abstract types applied to arguments. [Text (args, s)] is
+      the application of the abstract type constructor [s] to
+      arguments [args]. *)
   | Tfarray of t * t
+  (** Functional arrays. [TFarray (src,dst)] maps values of type [src]
+      to values of type [dst]. *)
   | Tnext of t
+  (** TODO: ? *)
   | Tsum of Hstring.t * Hstring.t list
+  (** Enumeration, with its name, and the list of its constructors. *)
   | Trecord of trecord
+  (** Record type. *)
+
+and tvar = {
+  v : int;
+  (** Unique identifier *)
+  mutable value : t option;
+  (** Pointer to the current value of the type variable. *)
+}
+(** Type variables.
+    The [value] field is mutated during unification,
+    hence distinct types should have disjoints sets of
+    type variables (see function {fresh}). *)
 
-and tvar = { v : int ; mutable value : t option }
 and trecord = {
   mutable args : t list;
+  (** parameters of the record
+      TODO: or are they the arguments applied
+            to the record constructor type ? *)
```

Comment:
they are type variables, like in `Text` case

Reply:
Aren't they rather the concrete types that replace the type variables ? (just as in the `Text` case)

Reply:
Yes